### ISSUE TITLE
Which version of proto-gen-go should I download when using gogo/protobuf@v1.0.0 because of gormv1?

### ISSUE THREAD

**Issue by daodennis** [gogo/protobuf#705]
Hi,

tl;dr How do I go install a version  of `proto-gen-go` that is compatible with specifically `gogo/protobuf@v1.0.0` and more importantly what is that version, how new or old must it be for 1.0.0?
Aka what is `???` in  `go get github.com/golang/protobuf/protoc-gen-go@` **`?????`**  It seems that that the version is pretty important according to https://github.com/golang/protobuf/issues/763#issuecomment-442434870

> CI scripts that hard-code a specific version of the proto package, but continues to obtain protoc-gen-go from HEAD is fundamentally not hermetic. (In fact, if any part of the CI script depends on anything from HEAD, then it is not hermetic). Breakages due to externally changing factors (e.g., 8d0c54c ) is an implicit (and undesirable) property of non-hermetic builds.

I am in an entanglement with `gogo/protobuf` because a project I am working on uses Gorm v1, which  uses `proto-gen-gorm` to generate some code and it depends on hard  `gogo/protobuf@v1.0.0.0` see this issue in `proto-gen-gorm` 

https://github.com/infobloxopen/protoc-gen-gorm/issues/174#issuecomment-530110900
> Sept '19 I think this is a symptom of the same issue as #98, I think. The Protobuf libraries (golang and gogo) make a > > big jump from v1.0.0 to v1.1.1 last year which deprecated a lot of functions used in this project. According to > > #98 (comment) doing a make vendor in the project repo can resolve some of the issues, but may also introduce other ones. I'd love to fix this, but it looked like it would require some pretty significant refactoring when I probed into it 
 last year. There may also be a simple solution to make it install with the proper dependencies, but I've always found dependency resolution confusing, so my hopes aren't high that I would figure it out.

**Comment by daodennis** [comment]
The one question I do have is how `protoc-gen-gorm` can have a dependency on `github.com/golang/protobuf v1.3.2` but only `github.com/gogo/protobuf v1.0.0`  and what the implications are for itself and other projects' build and runtime dependencies are.

**Comment by daodennis** [comment]
original bug opener here- since protobuf v1 and gormv1 are pretty outdated I am going to close this especially since it has been almost a couple of years anyway.
